sdks/python/apache_beam/runners/interactive/background_caching_job.py

```diff
@@ -19,29 +19,113 @@
 
 For internal use only; no backwards-compatibility guarantees.
 
-A background caching job is a job that caches events for all unbounded sources
-of a given pipeline. With Interactive Beam, one such job is started when a
-pipeline run happens (which produces a main job in contrast to the background
+A background caching job is a job that captures events for all capturable
+sources of a given pipeline. With Interactive Beam, one such job is started when
+a pipeline run happens (which produces a main job in contrast to the background
 caching job) and meets the following conditions:
 
-  #. The pipeline contains unbounded sources.
+  #. The pipeline contains capturable sources, configured through
+     interactive_beam.options.capturable_sources.
   #. No such background job is running.
   #. No such background job has completed successfully and the cached events are
-     still valid (invalidated when unbounded sources change in the pipeline).
+     still valid (invalidated when capturable sources change in the pipeline).
 
 Once started, the background caching job runs asynchronously until it hits some
-cache size limit. Meanwhile, the main job and future main jobs from the pipeline
-will run using the deterministic replay-able cached events until they are
-invalidated.
+capture limit configured in interactive_beam.options. Meanwhile, the main job
+and future main jobs from the pipeline will run using the deterministic
+replayable captured events until they are invalidated.
 """
 
 # pytype: skip-file
 
 from __future__ import absolute_import
 
+import logging
+import threading
+import time
+
 import apache_beam as beam
-from apache_beam import runners
 from apache_beam.runners.interactive import interactive_environment as ie
+from apache_beam.runners.interactive.caching import streaming_cache
+from apache_beam.runners.runner import PipelineState
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
```

Comment:
We plan to log at INFO level for this module.
Since each module in Beam runners has its own logger (since [PR](https://github.com/apache/beam/commit/49d6efdd2a59652462228e3e2b353bcc4173554b)), this overriding is intended and has limited effect.

Reply:
It has limited effect but it still overrides user choice. Presumably we can have a pipeline option that can set logging level per module level.

Also, is not the default logging level info anway?

Reply:
As tested, the default logging level in a notebook is `warning` level.

Reply:
I can get rid of the `setLevel` and revisit this when we have a non-logging display utility implemented for anything `is_in_notebook`.

Reply:
Removed the logging level setting and added a TODO to replace logging with ipython `display` when `is_in_notebook`.